src/ccs/multifolding.rs

```diff
@@ -25,58 +25,108 @@ use core::{cmp::max, marker::PhantomData};
 use ff::{Field, PrimeField};
 use flate2::{write::ZlibEncoder, Compression};
 use itertools::concat;
+use rand_core::RngCore;
 use rayon::prelude::*;
 use serde::{Deserialize, Serialize};
 use sha3::{Digest, Sha3_256};
 use std::ops::{Add, Mul};
 use std::sync::Arc;
 
-// XXX: THe idea is to have Multifolding as IVC instance in the future, holding the main CCS
-// instances. Then the rest of CCS, CCCS, LCCCS hold references to it.
-// Is our single source of data.
+/// The NIMFS structure is the center of operations of the folding scheme.
+/// Once generated, it allows us to fold any upcomming CCCS instances within it without needing to do much.
+// XXX: Pending to add doc examples.
 #[derive(Debug)]
-pub struct Multifolding<G: Group> {
-  ccs: CCSShape<G>,
+pub struct NIMFS<G: Group> {
+  ccs: CCS<G>,
   ccs_mle: Vec<MultilinearPolynomial<G::Scalar>>,
+  ck: CommitmentKey<G>,
+  lcccs: LCCCS<G>,
 }
 
-impl<G: Group> Multifolding<G> {
-  /// Generates a new Multifolding instance based on the given CCS.
-  pub fn new(ccs: CCSShape<G>) -> Self {
-    let ccs_mle = ccs.M.iter().map(|matrix| matrix.to_mle()).collect();
-    Self { ccs, ccs_mle }
+impl<G: Group> NIMFS<G> {
+  /// Generates a new NIMFS instance based on the given CCS.
```

Comment:
It seems like it takes more than just ccs

Reply:
Fixed in 52f6a62